### PR TITLE
Include multi search

### PR DESCRIPTION
**Pull request by GiannottiRod**
Cria a função busca_multi, que faz a procura por multiplos CEPs,
respeitando o intervalo mínimo de 1 segundo entre cada procura, e
retorna as informações encontradas em uma tibble. Para que ela funcione
pequenas mudanças na função parse.api foram feitas para que as colunas
da tibble criada sempre fossem criadas com tipos de variáveis coerentes
(similar ao que já se havia feito nas colunas latitude e longitude)

### CODE REVIEW

File: R/busca_multi.R [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+#' @importFrom magrittr %>%
+#' @importFrom magrittr %<>%
+#' @importFrom dplyr add_row
+#' @param lista_ceps lista de CEPs (postal code list).
+#' @param token Token de autorização. Veja <http://cepaberto.com/users/register>.
+#' @export
+
+busca_multi <- function( lista_ceps = c("01001000", "01001001") , token = NULL ) {
+  CEPS <- parse_api(NA) %>% na.exclude()
+  lista_ceps %<>%  as.vector()
```
Comment (RobertMyles) [inline R/busca_multi.R:12]: Is `as.vector()` necessary here? I get:

```r
> lista_ceps = c("01001000", "01001001")
> identical(lista_ceps, as.vector(lista_ceps))
[1] TRUE
```

File: R/busca_multi.R [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  for (i in lista_ceps) {
```
Comment (RobertMyles) [inline R/busca_multi.R:13]: Also getting this when running `busca_multi()`:

```r
> busca_multi(token = token)
Error: Can't combine `..1$ddd` <character> and `..2$ddd` <integer>.
Run `rlang::last_error()` to see where the error occurred.
> rlang::last_error()
<error/vctrs_error_incompatible_type>
Can't combine `..1$ddd` <character> and `..2$ddd` <integer>.
Backtrace:
  1. global::busca_multi(token = token)
  2. [ `%<>%`(...) ] with 7 more calls
 10. tibble::add_row(., busca_cep(cep = i, token = token))
 11. tibble:::rbind_at(.data, df, pos)
 12. vctrs::vec_rbind(old, new)
 14. vctrs::vec_default_ptype2(...)
 15. vctrs::stop_incompatible_type(...)
 16. vctrs:::stop_incompatible(...)
 17. vctrs:::stop_vctrs(...)
Run `rlang::last_trace()` to see the full context.
> rlang::last_trace()
<error/vctrs_error_incompatible_type>
Can't combine `..1$ddd` <character> and `..2$ddd` <integer>.
Backtrace:
     █
  1. └─global::busca_multi(token = token)
  2.   └─CEPS %<>% add_row(busca_cep(cep = i, token = token))
  3.     ├─base::withVisible(eval(quote(`_fseq`(`_lhs`)), env, env))
  4.     └─base::eval(quote(`_fseq`(`_lhs`)), env, env)
  5.       └─base::eval(quote(`_fseq`(`_lhs`)), env, env)
  6.         └─`_fseq`(`_lhs`)
  7.           └─magrittr::freduce(value, `_function_list`)
  8.             ├─base::withVisible(function_list[[k]](value))
  9.             └─function_list[[k]](value)
 10.               └─tibble::add_row(., busca_cep(cep = i, token = token))
 11.                 └─tibble:::rbind_at(.data, df, pos)
 12.                   └─vctrs::vec_rbind(old, new)
 13.                     └─(function () ...
 14.                       └─vctrs::vec_default_ptype2(...)
 15.                         └─vctrs::stop_incompatible_type(...)
 16.                           └─vctrs:::stop_incompatible(...)
 17.                             └─vctrs:::stop_vctrs(...)
```

**Review by RobertMyles — COMMENTED** [review]
Hey Rodrigo,

Thanks for your PR, I like the addition of this new function, looks great. I just have a few doubts, see the comments in R/busca_multi.R

Thanks,
Rob

(strike these, working fine now -- Rob)

File: R/busca_multi.R (hunk 1 above)
Comment (GiannottiRod) [reply R/busca_multi.R:12]: I was having problems using data from tibbles directly into the function - e.g. I have a dataset with names of stores and their CEP and want an approximate latitude and longitude, so ideally my code would just include dataset$CEP %>% busca_multi() -, and this solved the problem by making sure that the user can either use data directly from a new vector or by choosing a column of a tibble or data frame.

File: R/busca_multi.R (hunk 2 above)
Comment (GiannottiRod) [reply R/busca_multi.R:13]: Regarding the error:
The changes I proposed to parse_api() - including specification of a format for possible NAs - were supposed to address it without changing any functionality of parse_api(). When running it from my end with the updated parse_api it works just fine.
image

the line 
```
CEPS <- parse_api(NA) %>% na.exclude()
```
Is indeed bothering me, but it was the simplest way to initialize an empty tibble for the loop I managed to write. I am not sure if there is a better way to do it, but this works fine and considering the mandatory 1 second pauses in between searches I was not too worried about minimal performance hits.

File: R/busca_multi.R (hunk 1 above)
Comment (RobertMyles) [reply R/busca_multi.R:12]: Sure, I understand. I think it's fine.

File: R/busca_multi.R (hunk 2 above)
Comment (RobertMyles) [reply R/busca_multi.R:13]: Yeah, it works fine for me now, the mistake was on my end. Regarding `na.exclude()`, I was wondering initially why you would do it that way but it makes perfect sense. I'm just tidying up some small things in the merge and I'll submit the new version to CRAN asap. Thanks for your contribution, Rodrigo!

**Comment by GiannottiRod** [comment]
Thank you,

If you need any more info from me please let me know.

Thanks again for writing this package it has helped me a lot.

**Comment by RobertMyles** [comment]
My pleasure, happy I could help. It's @edumucelli and the CEPAberto project who deserve the real praise!

**Comment by RobertMyles** [comment]
Ah, one more thing, @GiannottiRod  -- would you mind updating the README with an example of `busca_multi()` whenever you get a chance? Thanks.
v0.1.2 has been submitted to CRAN, should be online soon hopefully.

**Comment by GiannottiRod** [comment]
Just sent another pr updating the README.Rmd I also fixed a small (literally one letter) Portuguese mistake in it.

**Pull request merged**